### PR TITLE
[Ready] Updated `raise` calls to `UI.user_error!` or `UI.crash!`

### PR DESCRIPTION
**Pull request by KrauseFx**
Oh yeah :rocket: 

All fastlane exceptions (except for spaceship ones) are now triggered using `UI.user_error!` or `UI.crash!`

**Old:**
screenshot 2016-03-24 11 47 09

**New:**
screenshot 2016-03-24 11 47 23

Code I used

``` ruby
task :converter do
  path = "./**/*.rb"
  Dir[path].each do |current|
    next if current.include?("spaceship/")
    content = File.read(current)
    ["'", '"'].each do |esc|
      other = (esc == "'" ? '"' : "'")
      content.gsub!(/raise #{esc}([\w\^\(\)\#\.\[\]\,\-\{\}\|\:\>\<\=\/\:\`\?\.\s#{other}]*)#{esc}(\.red){0,1}/) do |line|
        error_message = Regexp.last_match[1]
        puts "error:"
        puts error_message
        puts "line:"
        puts line
        "UI.user_error!(\"#{error_message}\")"
      end

      # end.to raise_error "Error message here".red
      content.gsub!(/end.to raise_error(.*)\.red/) do |line|
        error_message = Regexp.last_match[1]
        line.gsub(".red", "")
      end
    end
    File.write(current, content)
    puts content
  end
end
```

### CODE REVIEW

File: fastlane/lib/fastlane/one_off.rb [hunk 1]
```diff
@@ -17,14 +17,14 @@ def self.execute(args: nil)
...
-      raise "invalid syntax" unless action_name
+      UI.user_error!("invalid syntax") unless action_name
 
       class_name = action_name.fastlane_class + 'Action'
       class_ref = nil
       begin
         class_ref = Fastlane::Actions.const_get(class_name)
       rescue NameError
-        raise "Action not found"
+        UI.user_error!("Action '#{class_name}' not found")
```
Comment (asfalcone) [inline fastlane/lib/fastlane/one_off.rb]: I wonder if this should be `crash!` instead? seems like you might want a trace of how you got here?

File: fastlane/lib/fastlane/one_off.rb [hunk 2]
```diff
@@ -17,14 +17,14 @@ def self.execute(args: nil)
         end
       end
 
-      raise "invalid syntax" unless action_name
+      UI.user_error!("invalid syntax") unless action_name
```
Comment (asfalcone) [inline fastlane/lib/fastlane/one_off.rb]: I wonder if this should be `crash!` instead? seems like you might want a trace of how you got here?

File: fastlane/lib/fastlane/runner.rb [hunk 3]
```diff
@@ -20,15 +20,15 @@ def full_lane_name
     # @param parameters [Hash] The parameters passed from the command line to the lane
     # rubocop:disable Metrics/AbcSize
     def execute(lane, platform = nil, parameters = nil)
-      raise "No lane given" unless lane
+      UI.user_error!("No lane given") unless lane
```
Comment (asfalcone) [inline fastlane/lib/fastlane/runner.rb]: I wonder if this should be `crash!` instead? seems like you might want a trace of how you got here?

File: fastlane_core/spec/configuration_file_spec.rb [hunk 4]
```diff
@@ -21,7 +21,7 @@
           FastlaneCore::ConfigItem.new(key: :app_identifier,
                                        description: "Mac",
                                        verify_block: proc do |value|
-                                         raise "Invalid identifier '#{value}'" unless value.split('.').count == 3
+                                         FastlaneCore::UI.user_error!("Invalid identifier '#{value}'") unless value.split('.').count == 3
```
Comment (asfalcone) [inline fastlane_core/spec/configuration_file_spec.rb]: Why is `FastlaneCore::` required here?

**Comment by asfalcone** [comment]
This looks awesome. Left a few comments about places that might be better as `.crash` mostly because the error message didn't contain a enough information that a user could figure out what to change easily or how that code was getting called.

File: fastlane_core/spec/configuration_file_spec.rb (hunk 4 above)
Comment (KrauseFx) [inline fastlane_core/spec/configuration_file_spec.rb]: Oh it's not, it's optional. I added this by mistake.

**Comment by KrauseFx** [comment]
Thanks @asfalcone, I updated the PR :+1:

**Comment by i2amsam** [comment]
:+1: :rocket: :roller_coaster: !

**Comment by KrauseFx** [comment]
Oh yeah!!! :rocket:

**Pull request merged**